sgkit/io/vcf/vcf_reader.py

```diff
@@ -434,13 +443,23 @@ def vcf_to_zarr_sequential(
                     max_variant_allele_length, max(len(x) for x in alleles)
                 )
 
+                variant_quality[i] = (
+                    variant.QUAL if variant.QUAL is not None else FLOAT32_MISSING
+                )
+                variant_filter[i] = (
```

Comment:
I'm not sure anyone really uses the FILTER field so I'm not overly bothered, but do we want to "unparse" this field like this? I guess it's something to do with the VlenUtf8 codec? Good to explain the rationale here and we'd definitely want to document that this is how the FILTER field is stored.

Reply:
Done

Reply:
Thinking about this more, perhaps it would be better to store this as an array of ints, which are offsets into a `filters` attribute - a bit like `variant_contig` being offsets into the `contigs` attribute.

Reply:
That would be nicer all right - save a bit of space too I'd imagine.

Reply:
I'm changed the implementation (and [draft spec](https://github.com/tomwhite/sgkit/blob/vcfzarr-spec/vcf_zarr_spec.md)) to encode filters in a boolean array of shape (variants, filters), where a true value in the `i`-th position means that the `i`-th filter is applied. A list of filter strings is also stored in the `filters` attribute.

How does this look?

Reply:
Yeeesssss... yes, you have to do it this way because we have have a number of filters, and we would have to have a ragged array for filters to represent a number of them. So, this is better in the common not-very-many-filters case. Would be worse if there was say, hundreds of filters, but I don't think this happens (much - I bet somebody is doing it though).

What happens if (when) someone passes a filter value that's not declared in the header? It's probably not entirely valid VCF, but I bet htslib will happily accept it.

Reply:
> Would be worse if there was say, hundreds of filters, but I don't think this happens (much - I bet somebody is doing it though).

These boolean arrays are very compressible though, so it shouldn't be a problem.

> What happens if (when) someone passes a filter value that's not declared in the header? It's probably not entirely valid VCF, but I bet htslib will happily accept it.

You are right: htslib does accept it. So we should probably handle it as we do a missing contig, by raising an error.

Reply:
SGTM!